supersede.go

```diff
@@ -0,0 +1,110 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"path/filepath"
+
+	"github.com/taskcluster/generic-worker/gwconfig"
+	"github.com/taskcluster/httpbackoff"
+	"github.com/taskcluster/taskcluster-base-go/scopes"
+)
+
+var (
+	supersededByPath = filepath.Join("generic-worker", "superseded-by.json")
+	supersededByName = "public/superseded-by.json"
+)
+
+type SupersedeFeature struct {
+}
+
+type SupersedesServiceResponse struct {
+	TaskIDs []string `json:"supersedes"`
+}
+
+func (feature *SupersedeFeature) Name() string {
+	return "Supersede"
+}
+
+func (feature *SupersedeFeature) Initialise() error {
+	return nil
+}
+
+func (feature *SupersedeFeature) PersistState() error {
+	return nil
+}
+
+// Supersede is always enabled
+func (feature *SupersedeFeature) IsEnabled(fl EnabledFeatures) bool {
+	return true
+}
+
+type SupersedeTask struct {
+	task *TaskRun
+}
+
+func (feature *SupersedeFeature) NewTaskFeature(task *TaskRun) TaskFeature {
+	task.featureArtifacts[supersededByName] = "superseded feature"
+	return &SupersedeTask{
+		task: task,
+	}
+}
+
+func (l *SupersedeTask) RequiredScopes() scopes.Required {
+	// let's not require any scopes, as I see no reason to control access to this feature
+	return scopes.Required{}
+}
+
+func (l *SupersedeTask) Start() *CommandExecutionError {
+	supersederURL := l.task.Payload.SupersederURL
+	if supersederURL == "" {
+		return nil
+	}
+	resp, _, err := httpbackoff.Get(supersederURL)
+	if err != nil {
+		return MalformedPayloadError(err)
+	}
+	decoder := json.NewDecoder(resp.Body)
+	var supersedes SupersedesServiceResponse
+	err = decoder.Decode(&supersedes)
+	if err != nil {
+		return MalformedPayloadError(err)
```

Comment:
This is a confusing status to return in either case (HTTP error or bad JSON), as the payload is not, in fact, malformed.  Instead, just don't supersede -- that's [what docker-worker does](https://github.com/taskcluster/docker-worker/blob/08876fcfa89d600dc2c824b28c0fe72021cf3594/src/lib/task_listener.js#L425).  In general, superseding is designed to fail open, so that an issue with the superseder URL doesn't cause a treeclosure.